warden/x/warden/types/v1beta3/space.go

```diff
@@ -14,21 +15,49 @@ func (w *Space) IsOwner(address string) bool {
 	return false
 }
 
-func (w *Space) AddOwner(address string) error {
+func (w *Space) AddOwner(address string, nonce uint64) error {
+	if err := w.EnsureNonce(nonce); err != nil {
+		return err
+	}
+
 	if w.IsOwner(address) {
 		return ErrDuplicateSpaceOwner
 	}
 	w.Owners = append(w.Owners, address)
 	return nil
 }
 
-func (w *Space) RemoveOwner(address string) {
+func (w *Space) RemoveOwner(address string, nonce uint64) error {
+	if err := w.EnsureNonce(nonce); err != nil {
+		return err
+	}
+
 	for i, owner := range w.Owners {
 		if owner == address {
 			w.Owners = append(w.Owners[:i], w.Owners[i+1:]...)
-			return
+			return nil
 		}
 	}
+
+	return nil
+}
+
+func (w *Space) EnsureNonce(nonce uint64) error {
+	if w.Nonce != nonce {
+		return fmt.Errorf("space nonce does not match expected nonce: %d", w.Nonce)
```

Comment:
pls register this error in `errors.go` so we can give it a unique id

Reply:
Fixed